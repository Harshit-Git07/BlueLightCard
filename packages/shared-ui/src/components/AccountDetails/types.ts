export type AccountDetailsProps = {
  accountNumber: string;
  firstName: string;
  lastName: string;
};
