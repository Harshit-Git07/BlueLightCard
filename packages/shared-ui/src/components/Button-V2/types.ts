import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ButtonProps = PropsWithChildren & {
  className?: string;
  type?: 'button' | 'submit';
  disabled?: boolean;
  slim?: boolean;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  newTab?: boolean;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
  size?: ButtonSize;
  'data-testid'?: string;
};

export type ButtonSize = 'XSmall' | 'Small' | 'Large';
