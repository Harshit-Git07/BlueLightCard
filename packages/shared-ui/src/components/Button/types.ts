import { MouseEventHandler, PropsWithChildren } from 'react';
import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ThemeVariant } from '../../types';

export type ColorToken = {
  bg: string;
  hover: string;
  focus: string;
  text: string;
  border: string;
};

export type ButtonProps = PropsWithChildren & {
  type?: 'button' | 'submit' | 'link';
  disabled?: boolean;
  slim?: boolean;
  className?: string;
  variant?: ThemeVariant;
  invertColor?: boolean;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  href?: string;
  onClick?: MouseEventHandler;
  withoutHover?: boolean;
  withoutFocus?: boolean;
  borderless?: boolean;
};
