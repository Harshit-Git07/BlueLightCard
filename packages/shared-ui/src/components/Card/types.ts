import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { RefObject } from 'react';

export type CardState = 'default' | 'hover' | 'selected';

export interface CardProps {
  initialCardState?: CardState;
  cardTitle?: string;
  description?: string;
  truncateDescription?: boolean
  buttonTitle?: string;
  imageAlt?: string;
  imageSrc?: string;
  iconLeft?: IconDefinition;
  iconRight?: IconDefinition;
  showImage?: boolean;
  showDescription?: boolean;
  showButton?: boolean;
  onClick?: () => void;
  ariaLabel?: string;
  className?: string;
  'data-testid'?: string;
  canHover?: boolean;
}

export interface CardContentProps {
  title: string;
  description?: string;
  showDescription?: boolean;
  centerContent?: boolean;
  ariaLabel?: string;
  descriptionRef: RefObject<HTMLParagraphElement>;
  truncateDescription?: boolean;
}
