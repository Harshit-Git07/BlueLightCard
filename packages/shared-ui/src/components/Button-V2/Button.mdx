import { Meta, Canvas, Controls } from '@storybook/blocks';

import * as Button from './Button.stories';

<Meta of={Button} />

# Button

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)
- [Variants](#variants)
- [Design](#design)

## Overview
Buttons are clickable items used for actions and to trigger events, for example, opening/closing a dialogue, cancelling an action or performing a delete operation or submitting a form. They can have different styles depending on their needs.

Tertiary buttons, which look similar to links, are used for less important or less commonly used actions.

## Props
<Controls />

## Usage
➡️ Buttons may contain icon, on the left or the right side, padding between icon and text is 8px.

➡️ Use 8 px spacing between buttons stacked horizontally and 16px space between buttons stacked vertically.

➡️ Replace text with a loading action is submitted, but still processing(Out of scope) 24px horizontal padding and fixed height based on each button size.

➡️ Button is always set to hug contents avoid changing width. In certain layouts button width fills container i.e mobile screen.

➡️ Use only one primary button per content area, and any remaining calls to action (CTA) should be represented as lower emphasis buttons.

➡️ Should have just 1 word for explaining the action; a maximum of 3 words where only the first one will be capitalised something

➡️ If two buttons are stacked horizontally, secondary action uses secondary button at min width and primary action fills containerMust include sub-section on accessibility

## Variants
List of variants can be shown below here.

### Primary
<Canvas of={Button.Primary} />

### Secondary
<Canvas of={Button.Secondary} />

### Tertiary
<Canvas of={Button.Tertiary} />

### PrimaryDanger
<Canvas of={Button.PrimaryDanger} />

### TertiaryDanger
<Canvas of={Button.TertiaryDanger} />

### Design
Please see the [Figma design](<https://www.figma.com/design/OHV7qGPnjsPLs5aTcYwSoB/Blue-Print-(WIP)?node-id=1605-12652&node-type=frame&t=zju6ZcPBgfJJmo3q-0>) for the design of the Button component.
