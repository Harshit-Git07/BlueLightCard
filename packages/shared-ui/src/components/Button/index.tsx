import { FC } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { ThemeColorTokens, ThemeVariant } from '../../types';
import { cssUtil } from '../../utils/cssUtil';
import { ButtonProps, ColorToken } from './types';

// Note: as part of Globalisation tokens, inverted buttons and outline style element are out of scope.

const color: ThemeColorTokens = {
  [ThemeVariant.Primary]: {
    base: {
      bg: 'bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark',
      hover:
        'hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark',
      focus: 'focus:outline-[#2EB8E6] dark:focus:outline-[#FFFF00]',
      text: 'text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark',
      border: 'border-transparent',
      disabled:
        'disabled:bg-button-primary-disabled-bg-colour-light disabled:dark:bg-button-primary-disabled-bg-colour-dark disabled:text-button-primary-disabled-label-colour-light disabled:dark:text-button-primary-disabled-label-colour',
    },
    invert: {
      bg: 'bg-[#FAFAFA]',
      hover: 'hover:bg-[#B3FAFAFA]',
      focus: 'focus:outline-0',
      text: 'text-[#000099]',
      border: 'border-transparent',
    },
  },
  [ThemeVariant.Secondary]: {
    base: {
      bg: '',
      hover:
        'hover:bg-button-secondary-hover-bg-colour-light dark:hover:bg-button-secondary-hover-bg-colour-dark hover:text-button-secondary-hover-label-colour-light dark:hover:text-button-secondary-hover-label-colour-dark',
      focus: 'focus:outline-[#2EB8E6] dark:focus:outline-[#FFFF00]',
      text: 'text-button-secondary-default-label-colour-light dark:text-button-secondary-default-label-colour-dark',
      border:
        'border-button-secondary-default-border-colour-light dark:border-button-secondary-default-border-colour-dark',
      disabled:
        'disabled:text-button-secondary-disabled-label-colour-light disabled:dark:text-button-secondary-disabled-label-colour-dark disabled:border-button-secondary-disabled-border-colour-light dark:disabled:border-button-secondary-disabled-border-colour-dark',
    },
    invert: {
      bg: 'bg-[rgba(0, 0, 0, 0)]',
      hover: 'hover:opacity-75',
      focus: 'focus:outline-0',
      text: 'text-[#FAFAFA]',
      border: 'border-[#FAFAFA]',
    },
  },
  [ThemeVariant.Tertiary]: {
    base: {
      hover:
        'hover:text-button-tertiary-hover-label-colour-light dark:hover:text-button-tertiary-hover-label-colour-dark',
      focus: 'focus:outline-[#2EB8E6] dark:focus:outline-[#FFFF00]',
      text: 'text-button-tertiary-default-label-colour-light dark:text-button-tertiary-default-label-colour-dark',
      border: '!border-transparent',
      disabled:
        'disabled:text-button-tertiary-disabled-label-colour-light disabled:dark:text-button-tertiary-disabled-label-colour-dark',
    },
  },
};

const Button: FC<ButtonProps> = ({
  iconLeft,
  iconRight,
  disabled,
  children,
  slim,
  className,
  invertColor,
  href,
  type = 'button',
  variant = ThemeVariant.Primary,
  onClick,
  withoutHover = false,
  withoutFocus = false,
  borderless = false,
}) => {
  const colorToken = (
    invertColor && color[variant].invert ? color[variant].invert : color[variant].base
  )! as ColorToken;

  const classes = cssUtil([
    type === 'link' ? 'inline-block' : '',
    slim ? 'py-1' : 'py-1.5',
    'px-5 rounded-md transition border-2',
    borderless ? '' : 'focus:outline outline-2 outline-offset-2',
    disabled ? colorToken.disabled : '',
    colorToken.text,
    !disabled && !withoutHover ? colorToken.hover : '',
    !withoutFocus ? colorToken.focus : '',
    colorToken.bg ?? '',
    colorToken.border ?? '',
    className ?? '',
  ]);

  const ButtonTag = type === 'link' ? 'a' : 'button';

  return (
    <>
      <ButtonTag
        href={href}
        type={type !== 'link' ? type : undefined}
        disabled={type !== 'link' ? disabled : undefined}
        className={`${classes}, text-button-label-font, font-button-label-font font-button-label-font-weight tracking-button-label-font leading-button-label-font`}
        onClick={type !== 'link' ? onClick : undefined}
      >
        {iconLeft && <FontAwesomeIcon className="mr-2" icon={iconLeft} />}
        {children}
        {iconRight && <FontAwesomeIcon className="ml-2" icon={iconRight} />}
      </ButtonTag>
    </>
  );
};

export default Button;
