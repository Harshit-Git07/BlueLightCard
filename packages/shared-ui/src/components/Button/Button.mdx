import { Meta, Canvas, Controls } from '@storybook/blocks';

import * as Button from './Button.stories';

<Meta of={Button} />

# Button

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)

## Overview
Overview description of the button component...

## Props
<Controls />

## Usage
➡️ Usage instruction one

➡️ Usage instruction two

➡️ Usage instruction three

➡️ Usage instruction four

## Variants
List of variants can be shown below here.

### Default
<Canvas of={Button.Default} />

### Primary
<Canvas of={Button.Primary} />

### Secondary
<Canvas of={Button.Secondary} />

### Tertiary
<Canvas of={Button.Tertiary} />

### Link
<Canvas of={Button.Link} />

### Primary Inverted
<Canvas of={Button.PrimaryInverted} />

### Secondary Inverted
<Canvas of={Button.SecondaryInverted} />
