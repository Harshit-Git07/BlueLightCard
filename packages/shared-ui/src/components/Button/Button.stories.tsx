import { Meta, StoryObj } from '@storybook/react';
import { PlatformVariant, SizeVariant } from '../../types';
import Button from './';

const meta: Meta<typeof Button> = {
  title: 'Button',
  component: Button,
};

type Story = StoryObj<typeof Button>;

export const Default: Story = {
  render: (args) => <Button {...args}>Button</Button>,
  args: {
    platform: PlatformVariant.Mobile,
    sizeVariant: SizeVariant.Medium,
  },
};

export default meta;
