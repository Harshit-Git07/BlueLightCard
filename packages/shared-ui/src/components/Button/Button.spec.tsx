import Button from './';
import { ButtonProps } from './types';
import { faMinus, faPlus } from '@fortawesome/pro-solid-svg-icons';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('Button component', () => {
  let props: ButtonProps;

  beforeEach(() => {
    props = {};
  });

  describe('smoke test', () => {
    it('should render component without error', () => {
      render(<Button {...props} />);

      const button = screen.getByRole('button');

      expect(button).toBeTruthy();
    });
  });

  describe('snapshot Test', () => {
    it('renders a button with the correct icon side "left"', () => {
      const component = renderer.create(<Button iconLeft={faMinus}>Button</Button>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders a button with the correct icon side "right"', () => {
      const component = renderer.create(<Button iconRight={faPlus}>Button</Button>);
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
