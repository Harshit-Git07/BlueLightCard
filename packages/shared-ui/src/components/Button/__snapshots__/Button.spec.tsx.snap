// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot Test renders a button with the correct icon side "left" 1`] = `
<button
  className=" py-1.5 px-5 rounded-md transition border-2 focus:outline outline-2 outline-offset-2  text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark focus:outline-[#2EB8E6] dark:focus:outline-[#FFFF00] bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark border-transparent , text-button-label-font, font-button-label-font font-button-label-font-weight tracking-button-label-font leading-button-label-font"
  type="button"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-minus mr-2"
    data-icon="minus"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={{}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M432 256c0 17.7-14.3 32-32 32L48 288c-17.7 0-32-14.3-32-32s14.3-32 32-32l352 0c17.7 0 32 14.3 32 32z"
      fill="currentColor"
      style={{}}
    />
  </svg>
  Button
</button>
`;

exports[`Button component snapshot Test renders a button with the correct icon side "right" 1`] = `
<button
  className=" py-1.5 px-5 rounded-md transition border-2 focus:outline outline-2 outline-offset-2  text-button-primary-default-label-colour-light dark:text-button-primary-default-label-colour-dark hover:bg-button-primary-hover-bg-colour-light dark:hover:bg-button-primary-hover-bg-colour-dark hover:text-button-primary-hover-label-colour-light dark:hover:text-button-primary-hover-label-colour-dark focus:outline-[#2EB8E6] dark:focus:outline-[#FFFF00] bg-button-primary-default-bg-colour-light dark:bg-button-primary-default-bg-colour-dark border-transparent , text-button-label-font, font-button-label-font font-button-label-font-weight tracking-button-label-font leading-button-label-font"
  type="button"
>
  Button
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-plus ml-2"
    data-icon="plus"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={{}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M256 80c0-17.7-14.3-32-32-32s-32 14.3-32 32V224H48c-17.7 0-32 14.3-32 32s14.3 32 32 32H192V432c0 17.7 14.3 32 32 32s32-14.3 32-32V288H400c17.7 0 32-14.3 32-32s-14.3-32-32-32H256V80z"
      fill="currentColor"
      style={{}}
    />
  </svg>
</button>
`;
